src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -146,11 +150,22 @@ ucc_tl_ucp_reduce_scatter_knomial_progress(ucc_coll_task_t *coll_task)
                 block_count, step_radix, local_seg_index);
             local_data  = PTR_OFFSET(sbuf, local_seg_offset * dt_size);
             reduce_data = task->reduce_scatter_kn.scratch;
-            if (UCC_OK != (status = ucc_dt_reduce_multi(
-                               local_data, rbuf, reduce_data,
-                               task->send_posted - p->iteration * (radix - 1),
-                               local_seg_count, local_seg_count * dt_size, dt,
-                               mem_type, args))) {
+            if (args->reduce.predefined_op == UCC_OP_AVG &&
+                (avg_pre_op ? ucc_knomial_pattern_loop_first_iteration(p)
```

Comment:
can we make a common tl/ucp level wrapper function for reduction to reduce code dupliction? 
E.g.
```
static inline ucc_status_t ucc_tl_ucp_reduce_multi(void *v1, void *v2, void *v3, size_t n_vectors, size_t count,
                                                                                size_t stride, ucc_datatype_t dt, ucc_memory_type_t mt, 
                                                                               ucc_coll_task_t *task, int is_avg) 
{
           if (is_avg) {
                      return ucc-dt_reduce_multi_alpha(..)
           } else {
                      return reduce_multi(,...)
           }
}
```

Then in algs:
```
int is_avg = (args->reduce.predefined_op == UCC_OP_AVG && (avg_pre_op ? ...))
ucc_tl_ucp_reduce_multi(... ,is_avg)
```